### PR TITLE
wip Feature/design preliminary sql schema

### PR DESCRIPTION
**Pull request by oskam**
**in progress:** magazine+library items models, replace sqlite to postgresql engine in TestConfig
* 3 model files: `books`, `library`, `users`
* fixtures for database testing
* `tests/populate.py` returns db models objects with filled fields

### CODE REVIEW

File: models/library.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+class RentalLog(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    book_copy_id = db.Column(db.Integer,
+                             db.ForeignKey('copy.id'),
+                             nullable=False)
+    user_id = db.Column(db.Integer,
+                        db.ForeignKey('users.id'),
+                        nullable=False)
+    borrow_time = db.Column(db.DateTime)
```
Comment (etude-ist) [inline models/library.py]: Add utc time handling there.

File: tests/populate.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
+from mimesis import Generic
+from models.users import User
+from models.books import Book, Copy, Author, Tag
+from models.library import RentalLog
+
+g = Generic('en')
+
+
+def populate_users(n=20, role=None):
```
Comment (etude-ist) [inline tests/populate.py]: Maybe this functions should be changed into fixtures.

File: models/books.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    __tablename__ = 'books'
+    id = db.Column(db.Integer, primary_key=True)
+    isbn = db.Column(db.String(128), unique=True)
+    authors = db.relationship('Author',
+                              secondary='books_authors',
+                              lazy='joined',
+                              backref=db.backref('books'))
+    title = db.Column(db.String(256))
+    original_title = db.Column(db.String(256))
+    table_of_contents = db.Column(db.String(256))
```
[the remark this answers is not in the document]
Comment (oskam) [reply models/books.py]: It was said before that Table of Content will be a photo, so I've designed it as to be path to the picture. But of course I can change it to be text, as you prefer.

File: models/books.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        return "<Copy: {} book_id={}>".format(self.asset_code, self.book_id)
+
+
+class Author(db.Model):
+
+    __tablename__ = 'authors'
+    id = db.Column(db.Integer, primary_key=True)
+    first_name = db.Column(db.String(64))
+    last_name = db.Column(db.String(64))
+
```
Comment (wojwarych) [inline models/books.py]: What if there is more than one author of a book or was 'edited by' someone?

File: models/books.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (etude-ist) [reply models/books.py]: Ah, ok. So for now don't change it, we will handle that later.

File: tests/populate.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+            tags=tags if tags else [],
+            description=description
+        )
+        books.append(book)
+    return books
+
+
+def populate_copies(book, n=35):
+    copies = []
+    while len(copies) < n:
```
Comment (etude-ist) [inline tests/populate.py]: I guess using comprehension list would be more pythonic.

File: models/books.py (hunk 4 above)
Comment (wojwarych) [reply models/books.py]: ok, got it, thx ;)

**Pull request closed without merging**